File: MotoFreaksBackend/src/main/java/com/MJ/MotoFreaksBackend/MotoFreaksBackend/resource/requests/NewChallengeModel.java
package com.MJ.MotoFreaksBackend.MotoFreaksBackend.resource.requests;

import com.MJ.MotoFreaksBackend.MotoFreaksBackend.models.QuestionAnswer;
import lombok.Data;

import java.util.List;

@Data
public class NewChallengeModel {
    private String name;
    private String company;
    private String model;
    private String generation;
    private List<QuestionAnswer> qaList;
}

File: src/app/modules/challenges/logic/dto/request/new-challenge.model.ts
import {QuestionAnswer} from "../response/question-answer.model";

export class NewChallengeModel {
  name: string
  general:boolean;
  company: string
  model: string
  generation: string
  qaList: QuestionAnswer[];


  constructor(name: string, general: boolean, company: string, model: string, generation: string, qaList: QuestionAnswer[]) {
    this.name = name;
    this.general = general;
    this.company = company;
    this.model = model;
    this.generation = generation;
    this.qaList = qaList;
  }
}
